launch/launch/actions/execute_process.py

```diff
@@ -382,6 +383,15 @@ def _shutdown_process(self, context, *, send_sigint):
         if self.__shutdown_received:
             # Do not handle shutdown more than once.
             return None
+
+        if self.process_details is None or self._subprocess_transport is None:
+            # Skip shutting down if the process is not ready
+            context.register_event_handler(
+                OnProcessStart(
+                    on_start=lambda event, context:
+                    self._shutdown_process(context, send_sigint=send_sigint)))
+            return None
+
         self.__shutdown_received = True
         if self.__completed_future is None:
```

Comment:
How is this conditional related to the one you're adding? From the comment:
> Execution not started so nothing to do

Is the trace in #418 appearing because the process is in an intermediate state between "not started" and "running"? Should the new conditional should be processed after this one?

Reply:
The trace in #418 appears because the process haven't started before the shutdown process and this attempts to send it a `SIGINT` signal.

With this new conditional the shutdown signal is not sent to processes not started or not available yet and sent after the events have started.

Although I've discussed this fix with @ivanpauno and he told me the correct approach would be making the shutdown process to check if a process haven't started and never start it all. But I didn't find and easy way to implement that.